code/datums/components/extensible_machine.dm

```diff
@@ -0,0 +1,78 @@
+/datum/component/extensible_machine
+	var/tool_behaviour
+	var/extension_speed
+	var/extension_directions
+	var/deployed_directions
+	var/list/extensions
+	var/list/obj/extended_objects = list()
+
+/datum/component/extensible_machine/Initialize(exst, ext_speed, tool, ext_dirs = ALL_CARDINALS)
```

Comment:
some documentation would be nice, just detailing what it does etc. Not like TG levels of doc, but some doc nontheless